addons/uom/models/uom_uom.py

```diff
@@ -93,6 +93,21 @@ def _check_category_reference_uniqueness(self):
             if uom_data['uom_count'] > 1:
                 raise ValidationError(_("UoM category %s should only have one reference unit of measure.") % (self.env['uom.category'].browse(uom_data['category_id']).name,))
 
+    @api.onchange('rounding')
+    def _onchange_rounding(self):
+        precision = self.env.ref('product.decimal_product_uom').digits
+        if self.rounding < 1.0 / 10.0**precision:
+            warning = {
+                    'title': _('Warning!'),
+                    'message':  _(
+                        "This rounding precision is higher than the Decimal Accuracy"
+                        " (%s digits).\nThis may cause inconsistencies in reservations.\n"
+                         "Please set a precision between %s and 1.")
```

Comment:
This message is very "warehouse-centric".  We can generalize it:
```suggestion
                        "This rounding precision is higher than the Decimal Accuracy"
                        " (%s digits).\nThis may cause inconsistencies in computations.\n"
                        "Please set a precision between %s and 1.")
```